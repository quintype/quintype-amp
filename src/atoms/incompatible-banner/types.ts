import { Story } from "../../types/story";

export interface IncompatibleBannerTypes {
  story: Story;
  inlineStyles?: object;
}
