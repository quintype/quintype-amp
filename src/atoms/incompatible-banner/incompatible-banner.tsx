import React from "react";
import styled from "styled-components";
import { withStory } from "../../context";
import get from "lodash.get";

const BannerWrapper = styled.div`
  position: absolute;
  top: 60px;
  background-color: ${(props) => props.theme.color.mono4};
  width: 100%;
  text-align: center;
  line-height: ${(props) => props.theme.font.lineHeight.level5};
  z-index: ${(props) => props.theme.zIndex.z1};
`;
const StyledBanner = styled.div`
  margin: 15px 10px;
`;

const IncompatibleBannerBase = ({ story }) => {
  const isAmpSupported = get(story, "is-amp-supported", null);
  return (
    !isAmpSupported && (
      <BannerWrapper>
        <StyledBanner>
          Some of the elements in this story are not compatible with AMP. To view the complete story, please{" "}
          <a href={story.url}>click here</a>
        </StyledBanner>
      </BannerWrapper>
    )
  );
};

export const IncompatibleBanner = withStory(IncompatibleBannerBase);
