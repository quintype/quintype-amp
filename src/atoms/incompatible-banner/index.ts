import { IncompatibleBanner } from "./incompatible-banner";
export { IncompatibleBanner };
