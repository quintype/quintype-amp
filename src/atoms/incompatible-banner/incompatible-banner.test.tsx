import React from "react";
import { StyledBanner, IncompatibleBanner } from "./incompatible-banner";
import { shallow } from "enzyme";

describe("IncompatibleBanner", () => {
  it("should render", () => {
    const wrapper = shallow(<IncompatibleBanner />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render styled banner with custom styles", () => {
    const wrapper = shallow(<StyledBanner inlineStyles={{ border: "5px solid red" }} />);
    expect(wrapper.find("div").prop("style")).toStrictEqual({ border: "5px solid red" });
  });
});
