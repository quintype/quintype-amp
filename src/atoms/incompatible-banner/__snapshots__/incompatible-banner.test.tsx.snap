// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncompatibleBanner should render 1`] = `
<IncompatibleBannerBase
  story={Object {}}
/>
`;
